stdlib/asm/math/u64.masm

```diff
@@ -630,7 +630,7 @@ end
 # The shift value is assumed to be in the range [0, 64).
 # Stack transition looks as follows:
 # [b, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = a << b mod 2^64.
-# This takes 13 cycles.
+# This takes 56 cycles.
```

Comment:
I wonder, if since this is an `unchecked` version, we should be using `pow2.unsafe` operation here and in the functions below.

Reply:
Got it, thanks! I will update it. Meanwhile, in `export.overflowing_shl` should we be using `pow2`?

Reply:
Yeah - I think in those we should use the `pow2`.

Reply:
`export.overflowing_shr` is using `unchecked_shr` & `unchecked_shl` & both of them are using `pow2.unsafe`. If we were to use `pow2` here then shall I add `safe` check operations before these two unchecked procs? I have not used any safe check for it in the current PR. Let me know if I should add the checks.  
https://github.com/maticnetwork/miden/blob/fd94f905c360fc6282cf0c9aeef89dc22df22a1e/stdlib/asm/math/u64.masm#L699-L712

Reply:
Actually, I take my last comment back. Based on the approach I laid out [here](https://github.com/maticnetwork/miden/issues/131#issuecomment-1082420829) and the "checked" variants check validity of inputs/outputs. All other variants are "unchecked".

Reply:
Thanks for sharing! I've removed the "safe" pow2 operations in the latest PR as none of the affected procs are "checked" variants. Please see [4c8a1b1](https://github.com/maticnetwork/miden/pull/284/commits/4c8a1b145b15fabe6c7277610f2f50ac326935a2).